motionplan/ik/nloptInverseKinematics.go

```diff
@@ -126,11 +160,16 @@ func (ik *NloptIK) Solve(ctx context.Context,
 
 		if len(gradient) > 0 {
```

Comment:
you shouldnt need to check this I think.  If gradient is length zero the loop below will give the same outcome